pkg/localmetrics/localmetrics.go

```diff
@@ -181,33 +181,27 @@ func AddCertificateIssuance(action string) {
 
 // UpdateCertValidDuration updates the Prometheus metric for certificate validity duration.
 func UpdateCertValidDuration(kubeClient client.Client, cert *x509.Certificate, now time.Time, clusterName, namespace string) {
-	if kubeClient == nil {
-		// If kubeClient is nil, set the metric value based on the certificate's expiration date
-		if cert != nil {
-			diff := cert.NotAfter.Sub(now)
-			days := math.Max(0, math.Round(diff.Hours()/24))
-			MetricCertValidDuration.With(prometheus.Labels{
-				"cn":      cert.Subject.CommonName,
-				"cluster": clusterName,
-			}).Set(days)
-		} else {
-			MetricCertValidDuration.With(prometheus.Labels{
-				"cn":      clusterName,
-				"cluster": clusterName,
-			}).Set(0)
-		}
+	var days float64
+	var cn string
+
+	// If both kubeClient and cert are nil, set to 0
+	if kubeClient == nil && cert == nil {
+		MetricCertValidDuration.With(prometheus.Labels{
+			"cn":      clusterName,
+			"cluster": clusterName,
+		}).Set(0)
 		return
 	}
 
-	// Check if the cluster is decommissioned
-	if isClusterDecommissioned(kubeClient, clusterName, namespace) {
-		logger.Info("Cluster is decommissioned, skipping UpdateCertValidDuration", "clusterName", clusterName)
-		return
+	// If kubeClient is available, check for decommissioning
+	if kubeClient != nil {
+		if isClusterDecommissioned(kubeClient, clusterName, namespace) {
+			logger.Info("Cluster is decommissioned, skipping metric update", "clusterName", clusterName)
+			return
```

Comment:
If a cluster is getting deleted, don't we need to adjust these metrics somehow? ie - set the gauge to zero to indicate the cert isn't valid anymore and/or delete it somehow?

Reply:
Currently, we are skipping the update for metrics of deleted or decommissioned clusters instead of setting them to 0 which was the ask from @iamkirkbater  earlier

Reply:
Yeah, explicitly setting it to 0 would result in alerts being fired, we'd need to just remove the metric with those cluster's labels entirely.

Skipping for now works, since they'll just never be updated (meaning they'll never decrement and not fire an alert) and re-deploying the operator would make them go away, but we could create a followup task to get the metrics to actually entirely be removed.